simtools/production_configuration/calculate_statistical_errors_grid_point.py

```diff
@@ -0,0 +1,460 @@
+
+        except FileNotFoundError as e:
+            error_message = f"Error loading file {self.file_path}: {e}"
+            _logger.error(error_message)
+            raise FileNotFoundError(error_message) from e
+        return data
+
+    def create_bin_edges(self):
+        """
+        Create unique energy bin edges.
+
+        Returns
+        -------
+        bin_edges : array
+            Array of unique energy bin edges.
+        """
+        bin_edges_low = self.data["bin_edges_low"]
+        bin_edges_high = self.data["bin_edges_high"]
+        bin_edges = np.concatenate([bin_edges_low, [bin_edges_high[-1]]])
+        return np.unique(bin_edges)
+
+    def compute_histogram(self, event_energies_reco, bin_edges):
+        """
+        Compute histogram for triggered events.
+
+        Parameters
+        ----------
+        event_energies_reco : array
+            Array of energies of the observed events.
+        bin_edges : array
+            Array of energy bin edges.
+
+        Returns
+        -------
+        triggered_event_histogram : array
+            Histogram of triggered events.
+        """
+        event_energies_reco = event_energies_reco.to(bin_edges.unit)
+
+        triggered_event_histogram, _ = np.histogram(event_energies_reco.value, bins=bin_edges.value)
+        return triggered_event_histogram * u.count
+
+    def compute_efficiency_and_errors(self, triggered_event_counts, simulated_event_counts):
+        """
+        Compute trigger efficiency and its statistical error using the binomial distribution.
+
+        Parameters
+        ----------
+        triggered_event_counts : array with units
+            Histogram counts of the triggered events.
+        simulated_event_counts : array with units
+            Histogram counts of the simulated events.
+
+        Returns
+        -------
+        efficiencies : array
+            Array of calculated efficiencies.
+        relative_errors : array
+            Array of relative uncertainties.
+        """
+        # Ensure the inputs have compatible units
+        triggered_event_counts = triggered_event_counts.to(u.ct)
+        simulated_event_counts = simulated_event_counts.to(u.ct)
+
+        # Compute efficiencies, ensuring the output is dimensionless
+        efficiencies = np.divide(
+            triggered_event_counts,
+            simulated_event_counts,
+            out=np.zeros_like(triggered_event_counts),
+            where=simulated_event_counts > 0,
+        ).to(u.dimensionless_unscaled)
+
+        # Set up a mask for valid data with a unit-consistent threshold
+        if np.any(triggered_event_counts > simulated_event_counts):
+            raise ValueError(
+                "Triggered event counts exceed simulated event counts. Please check input data."
+            )
+        valid = (simulated_event_counts > 0 * u.ct) & (triggered_event_counts > 0 * u.ct)
+
+        uncertainties = np.zeros_like(triggered_event_counts) * u.ct**-0.5
```

Comment:
Can you explain the '-0.5'?

Reply:
good point, this was wrongly implemented and the errors should be dimensionless. Previously when keeping the units the rel error turns out with this dimension.